WordPress/gutenberg
# Global Styles Sidebar: adjust icon vertical alignment with labels

## What?


This PR adjusts the vertical alignment between icons and labels in the Global Styles sidebar, as flagged in #38934.

- Follow-up: review the docs and examples for `HStack`, `Flex` and `ItemGroup`, in case some of the anti-patterns described below are used. 

## Why?


This is in order to follow the design specs highlighted in #38934 

## How?


After inspecting the layout of icons and labels, I noticed that the browser was always adding, under each icon, a few px of white space — this was due to the fact that the icons are an inline-block element within a block element, and therefore that white space was automatically added by the browser. Notice how the icon's wrapper element has a computed height of `27px`, instead of the expected `24px`:

  <img src="https://user-images.githubusercontent.com/1083581/161312467-0351666f-7147-4531-a56b-c0b92ffc3109.png"
       alt="The parent element of the icon is inspected in the browser and has a height of `27px`, instead of the expected `24px`">

This is a "classic" CSS layout problem (one hacky way to fix it, for example, is to set the `font-size` of the parent container to `0`). Instead, I spent some more time in order to come up with a better fix:

- The main fix was to remove the `FlexItem`/`FlexBlock` wrappers around `Icon` components (`FlexItem`/`FlexBlock` have `display: block`, and were not really being of any use. By removing them, the new parent element of the icon becomes the `display: flex` parent, which automatically ignores the white space that previously was causing the misalignment)
- Talking about `display: flex` parents, I had to apply those styles to the `ZStack` component to add that same fix the color palette screen (I will merge the changes to `ZStack` separately, but for now I've kept those changes within this PR for ease of understanding)
- A consequence of this change is that now, the items in the Color palette screen have become `4px` shorter with respect to all other items. This is because `<ColorIndicator />` has a size of `20px`, while the rest of the icons have a size of `24px`. To make up for this, I've decided to apply a `2px` top and bottom margin around the `<ColorIndicator />`s, but we could also look into adding the possibility for `<ColorIndicator />` to have a size of `24px` instead **(this needs design feedback)**

## Testing Instructions





- Select a theme that supports Full Site Editing
- Visit the site editor (from any WPAdmin screen, select "Appearance" and then "Editor" in the WPAdmin menu on the left)
- Carefully inspect icons across all Global Styles screens, and make sure that they are vertically aligned with the text label as expected (per the design spec, see #38934 )

## Screenshots or screencast 

Example of the changes on the root screen:

| Before | After |
|---|---|
| ![Screenshot 2022-04-01 at 20 40 24](https://user-images.githubusercontent.com/1083581/161323185-7a15670f-1cf0-4103-8038-d610ef3f2041.png) | ![Screenshot 2022-04-01 at 20 41 01](https://user-images.githubusercontent.com/1083581/161323386-df08a7fd-7fbe-4022-809d-f7284efce86f.png) |
